lib/puppet/provider/oneview_resource.rb

```diff
@@ -74,75 +66,79 @@ def data
     end
 
     def exists?(states = [nil, :found])
-      @data = data_parse
-      empty_data_check(states)
-      !@resource_type.find_by(@client, @data).empty?
+      prepare_environment
+      @item = @resource_type.new(@client, @data)
+      return true if empty_data_check(states)
+      return false unless @item.retrieve! && @item.like?(@data) && !@data['new_name']
+      Puppet.debug "#{@resource_type} #{@item['name']} is up to date."
+      true
       # @property_hash[:ensure] == :present # TODO: Future Improvement: Look into using property_hash for verifying existance globally
     end
 
-    # TODO: Would be awesome to have this working for everything/most types. Future improvement. Leaving as is in the meanwhile for filler.
     def create(action = :create)
       return true if resource_update
       ov_resource = if action == :create
                       @resource_type.new(@client, @data).create
                     elsif action == :add
                       @resource_type.new(@client, @data).add
+                    elsif action == :update
+                      raise 'This resource relies on others to be created.'
```

Comment:
At the moment, only using it for connection templates. There are likely other use cases though.